File: Problems/0290_Word_Pattern/Word_Pattern.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public bool WordPattern(string pattern, string str)
    {
        string[] words = str.Split(' ');

        if (pattern.Length != words.Length)
            return false;

        int i, j;

        for (i = 0; i < pattern.Length; ++i)
            for ( j = i + 1; j < pattern.Length; ++j)
                if (pattern[i] == pattern[j])
                    if (words[i] != words[j])
                        return false;
        
        for (i = 0; i < words.Length - 1; ++i)
            for (j = i + 1; j < words.Length; ++j)
                if (words[i] == words[j])
                    if (pattern[i] != pattern[j])
                        return false;

        return true;
    }

    public void Main(string args)
    {
        string[] flds = args.Replace("\"","").Split(',');
        string pattern = flds[0];
        string var_str = flds[1];

        Console.WriteLine("pattern = " + pattern + ", str = " + var_str);

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        bool result = WordPattern(pattern, var_str);

        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms");
    }
}

File: Problems/0200_0299/0290_Word_Pattern/Project_Python3/Word_Pattern.py
# coding: utf-8

import os
import sys
import time
import copy

class Solution:
    def wordPattern(self, pattern, str):
        """
        :type pattern: str
        :type str: str
        :rtype: bool
        """
        words = str.split(" ")

        if len(pattern) != len(words):
            return False

        for i in range(len(pattern)):
            for j in range(i + 1, len(pattern)):
                if pattern[i] == pattern[j]:
                    if words[i] != words[j]:
                        return False

        for i in range(len(words) - 1):
            for j in range(i + 1, len(words)):
                if words[i] == words[j]:
                    if pattern[i] != pattern[j]:
                        return False
        return True

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    tempStr = temp.replace("\"","").rstrip()

    flds = tempStr.split(',')
    pattern = flds[0]
    var_str = flds[1]

    print("pattern = {0}, str = {1}".format(pattern, var_str))

    sl = Solution()
    time0 = time.time()

    result = sl.wordPattern(pattern, var_str)

    time1 = time.time()

    print("result = {0}".format(result))
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
